### PR TITLE
feat: add support for JSON with comments `turbo.json` file

### LINKED ISSUE #644: Consider using json5 parser for `turbo.json`
### Describe the feature you'd like to request

I would like to add inline comments to the `turbo.json` file to explain somethings for later reference

### Describe the solution you'd like

Switch to a `json5` parser in Go which supports comments, e.g. https://github.com/barney-ci/go-json5?ref=golangexample.com

### Describe alternatives you've considered

Not use comments

### PR DESCRIPTION
**Pull request by weyert**
Allow the `turbo.json` file to contain comments

Makes it easier to explain the pipeline when they are quite large :)
So that future @weyert and his colleagues know what's going on!

Fixes #644

### CODE REVIEW

**Comment by weyert** [comment]
@gsoltis I am not sure why the benchmark build step is failing

File: cli/internal/fs/package_json.go [hunk 1]
```diff
@@ -22,12 +25,18 @@ type TurboConfigJSON struct {
 }
 
 func ReadTurboConfigJSON(path string) (*TurboConfigJSON, error) {
-	b, err := ioutil.ReadFile(path)
+	// b, err := ioutil.ReadFile(path)
```
Comment (jaredpalmer) [inline cli/internal/fs/package_json.go:28]: Remove comments

**Review by jaredpalmer — CHANGES_REQUESTED** [review]

File: cli/internal/fs/package_json.go (hunk 1 above)
Comment (weyert) [reply cli/internal/fs/package_json.go]: Removed them

**Review by gsoltis — APPROVED** [review]

**Comment by weyert** [comment]
Any more changes needed to get it merged? @jaredpalmer @gsoltis

**Review by jaredpalmer — APPROVED** [review]
I'm fine with this, but when we move to Rust, hopefully the json5 lib we choose will be close as to spec as this library is.

**Pull request merged**